keras_cv/layers/preprocessing/repeated_augmentation.py

```diff
@@ -0,0 +1,115 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+from keras_cv.layers.preprocessing.base_image_augmentation_layer import (
+    BaseImageAugmentationLayer,
+)
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class RepeatedAugmentation(BaseImageAugmentationLayer):
+    """RepeatedAugmentation augments each image in a batch multiple times.
+
+        This technique exists to emulate the behavior of stochastic gradient descent within
+        the context of mini-batch gradient descent.  When training large vision models,
+        choosing a large batch size can introduce too much noise into aggregated gradients
+        causing the overall batch's gradients to be less effective than gradients produced
+        using smaller gradients.  RepeatedAugmentation handles this by re-using the same
+        image multiple times within a batch creating correlated samples.
+
+        Args:
+            augmenters: the augmenters to use to augment the image
+            shuffle: whether or not to shuffle the result.  Essential when using an
+                asynchronous distribution strategy such as ParameterServerStrategy.
+
+        Usage:
+        ```python
+        repeated_augment = cv_layers.RepeatedAugmentation(
+            augmenters=[
+                cv_layers.RandAugment(value_range=(0, 255)),
+                cv_layers.RandomFlip(),
+            ]
+        )
+        inputs = {
+            "images": tf.ones((8, 512, 512, 3)),
+            "labels": tf.ones((8,)),
+        }
+        outputs = repeated_augment(inputs)
+        ```
+
+        References:
+        - [DEIT implementaton](https://github.com/facebookresearch/deit/blob/ee8893c8063f6937fec7096e47ba324c206e22b9/samplers.py#L8
+    )
+        - [Original publication](https://openaccess.thecvf.com/content_CVPR_2020/papers/Hoffer_Augment_Your_Batch_Improving_Generalization_Through_Instance_Repetition_CVPR_2020_paper.pdf)
+
+    """
+
+    def __init__(self, augmenters, shuffle=True, **kwargs):
+        super().__init__(**kwargs)
+        self.augmenters = augmenters
+        self.shuffle = shuffle
+
+    def _batch_augment(self, inputs):
+        images = inputs.get("images", None)
+        labels = inputs.get("labels", None)
+
+        if sorted(inputs.keys()) != ["images", "labels"]:
+            raise ValueError(
+                "RepeatedAugmentation() does not yet support tasks other than "
+                "classification."
+            )
+
+        if images is None or labels is None:
+            raise ValueError(
+                "RepeatedAugmentation expects inputs in a dictionary with format "
+                '{"images": images, "labels": labels}.'
+                f"Got: inputs = {inputs}"
+            )
+
+        image_results = []
+        labels_results = []
+
+        for augmenter in self.augmenters:
```

Comment:
I may be misunderstanding here, but can't we do away with some of the special-casing keys here with something like

```
augmenter_outputs = [augmenter(inputs) for augmenter in self.augmenters]

outputs = {}
for k in input.keys():
    outputs[k] = tf.concat([batch[k] for batch in augmenter_outputs, axis=0)
```

Reply:
doesn't work for bounding boxes or any other custom datatypes:

```
tf.concat(batch['bounding_boxes'])
# => cannot concatenate a dictionary object
```

Perhaps we ship it with special casing for bounding boxes but other than that just use:

```
outputs = {}
for k in input.keys():
    outputs[k] = tf.concat([batch[k] for batch in augmenter_outputs, axis=0)
```
?